### ISSUE TITLE
What about support for Proton Drive?

### ISSUE THREAD

**Issue by xinnir** [PhilippC/keepass2android#2313]
Is it not time to provide direct support for Proton Drive?

**Comment by Spiralpine** [comment]
+1

**Comment by PhilippC** [comment]
I don't think that Proton Drive exposes a suitable API.

**Comment by Quintovir** [comment]
+1 if possible.

**Comment by WalterPinkman** [comment]
I'd love to see this implemented. 

@PhilippC, do you mind explaining what is required from Proton's side so that their paying customers may request it?

**Comment by MateuszKubuszok** [comment]
As a fellow paying Proton customer I can answer: you can go to their [feature request page](https://protonmail.uservoice.com/forums/932839-proton-drive) and vote on relevant tickets e.g. [this one](https://protonmail.uservoice.com/forums/932839-proton-drive/suggestions/43582125-protondrive-api).

However, I am rather pessimistic as Proton refuses to work on polishing their existing product and instead every two months they release yet another half backed new one e.g. ProtonPass which has less features than any existing paid - or free! - password manager, and now Proton Captcha. I can only imagine that they wanted to capitalize on this year's passwords leaks but instead of supporting initiatives like KeePass by fixing their unusable Proton Drive (no local sync/automatic backup ability on Android, no API to access my own files through my other apps, no MacOS nor Linux desktop client) they pushed their new mobile or browser-only password manager.

**Comment by gerald2545** [comment]
hi,
a plugin exists now for keepas to access kdbx file from proton drive (https://github.com/dhaven/ProtonDriveSync#readme, referenced in keepass plugins list), would it be possible to implement it in keepass2android?

Thank you

**Comment by PhilippC** [comment]
yes, looks like this can be used to build a FileStorage implementation in KeePass2Android. Given the (not existing) number of up-votes here, I don't give this high priority, but I'm happy to accept any PR of course.

**Comment by Decoherent** [comment]
Well, I'll toss a vote in :) I hadn't realized that Proton Drive had developed into basically a drop-in replacement for OneDrive, except, of course, for privacy. It'd be neat!

**Comment by Soldier1st** [comment]
+1 if possible

**Comment by PhilippC** [comment]
please don't add "+1" comments but click the "+1" reaction on the top post. I use this to rate feature requests (https://github.com/PhilippC/keepass2android/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)

**Comment by recoder-lab** [comment]
Looks like this is supported in KeePassDX app.   Not sure if this will help with implementation

**Comment by R-Chandra** [comment]
Possible workaround: Before Keepass2Android had direct support for pCloud, I used the system file picker.  As long as my device has the pCloud app installed, pCloud shows up in the "hamburger menu."  (So does "OneDrive")  After tapping on that row, I can tap on directories (folders) until I get to where I store my databases.  As I'm not a Proton Drive customer, I don't know if its mobile app supports the same sort of things as the pCloud app, but I thought I'd put the suggestion out there.

**Comment by Decoherent** [comment]
Yep, that basically works OK. Things can get a little "off", given that PD caches data in a very strange way sometimes, but it generally does what you'd expect as a storage provider for Keepass2Android.

The Android Proton Drive app itself, though, is...primitive, perhaps, to be polite? It looks & acts like a "Summer of Code" project. It's missing some surprising functionality, like _search_. I'm a big Proton fan, and I use them exclusively in the domains where they operate, but I have no idea what their 500 employees do with their time.

**Comment by TomAFrench** [comment]
Hey, I'd also appreciate this feature request.

I saw someone posted an unofficial plugin above but Proton's starting to move towards publishing a C# SDK for proton drive https://proton.me/blog/proton-drive-sdk-preview

https://github.com/ProtonDriveApps/sdk-tech-demo

**Comment by PhilippC** [comment]
That actually seems like a great step forward. Thanks for the hint!

**Comment by mehmetalisahin** [comment]
+1

**Comment by antgustech** [comment]
SDK is out now at least but not quite ready for third party use according to readme.


https://github.com/ProtonDriveApps/sdk
